### PR TITLE
Make workflow cancellation idempotent

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Workflow cancellation is not idempotent
  1. request cancel on running workflow will return nil
  2. request cancel on finished workflow will return nil
  3. request cancel on non-existence workflow will return not found error

**Why?**
Make the request cancel logic idempotent

**How did you test it?**
Update & simplify integration tests

**Potential risks**
N/A

### CODE REVIEW

File: service/history/mutableStateBuilder.go [hunk 1]
```diff
@@ -1573,12 +1573,8 @@ func (e *mutableStateBuilder) IsWorkflowExecutionRunning() bool {
 	}
 }
 
-func (e *mutableStateBuilder) IsCancelRequested() (bool, string) {
-	if e.executionInfo.CancelRequested {
-		return e.executionInfo.CancelRequested, e.executionState.CreateRequestId
```
Comment (samarabbas) [inline service/history/mutableStateBuilder.go:1578]: Why were we using `CreateRequestId` as `CancelRequestID`?

File: service/history/historyEngine.go [hunk 2]
```diff
@@ -1767,18 +1769,10 @@ func (e *historyEngineImpl) RequestCancelWorkflowExecution(
...
-					}
-				}
-				// if we consider workflow cancellation idempotent, then this error is redundant
-				// this error maybe useful if this API is invoked by external, not workflow task from transfer queue.
-				return nil, ErrCancellationAlreadyRequested
+				// since cancellation is idempotent
+				return &updateWorkflowAction{noop: true}, nil
 			}
 
 			if _, err := mutableState.AddWorkflowExecutionCancelRequestedEvent("", req); err != nil {
```
Comment (samarabbas) [inline service/history/historyEngine.go:1778]: Probably not relevant to your change, but I think we should remove cause from `AddWorkflowExecutionCancelRequestedEvent`.  It does not makes sense for requestCancel.

**Review by samarabbas — APPROVED** [review]

File: service/history/historyEngine.go (hunk 2 above)
Comment (wxing1292) [reply service/history/historyEngine.go:1778]: Let me make a PR for this, probably tomorrow

Comment (wxing1292) [reply service/history/historyEngine.go:1778]: #957

**Pull request merged**